test/data_generation.jl

```diff
@@ -0,0 +1,14 @@
+@testset "Data Generation" begin
+    @testset "Generate Well Conditioned Matrix - RNG" begin
+        rng = MersenneTwister(1)
+        matrix = generate_well_conditioned_matrix(rng, 5)
+
+        @test isempty(matrix) == false
```

Comment:
let's test the "well conditioned" property (here and below too)... which should probably involve some test using `cond`, but i think testing that the matrix is positive definite is probably sufficient

```suggestion
        @test !isempty(matrix)
        @test isposdef(matrix)
```